cmd/kubermatic-installer/cmd_mirror_images.go

```diff
@@ -115,197 +126,225 @@ func MirrorImagesFunc(logger *logrus.Logger, versions kubermaticversion.Versions
+				// Otherwise some charts would fail to render (e.g. consul).
+				// Since we are just rendering from the client-side, it makes sense to use the latest kubeVersion we support.
+				latestClusterVersion := clusterVersions[len(clusterVersions)-1]
+				images, err := images.GetImagesForHelmCharts(ctx, chartsLogger, kubermaticConfig, helmClient, options.ChartsDirectory, options.HelmValuesFile, options.RegistryPrefix, latestClusterVersion.Version.Original())
 				if err != nil {
-					return fmt.Errorf("failed to create local addons path: %w", err)
+					return fmt.Errorf("failed to get images: %w", err)
 				}
-				defer os.RemoveAll(tempDir)
+				imageSet.Insert(images...)
+			}
 
-				options.AddonsPath = tempDir
+			logger.Info("🚀 Rendering system Applications Helm charts…")
+			appImages, err := images.GetImagesFromSystemApplicationDefinitions(logger, kubermaticConfig, helmClient, options.HelmTimeout, options.RegistryPrefix)
+			if err != nil {
+				return fmt.Errorf("failed to get images for system Applications: %w", err)
 			}
-		}
+			imageSet.Insert(appImages...)
 
-		logger.Info("🚀 Collecting images…")
-
-		// Using a set here for deduplication
-		imageSet := sets.New[string]()
-		for _, clusterVersion := range clusterVersions {
-			for _, cloudSpec := range images.GetCloudSpecs() {
-				for _, cniPlugin := range images.GetCNIPlugins() {
-					versionLogger := logger.WithFields(logrus.Fields{
-						"version":     clusterVersion.Version.String(),
-						"provider":    cloudSpec.ProviderName,
-						"cni-plugin":  string(cniPlugin.Type),
-						"cni-version": cniPlugin.Version,
-					},
-					)
-
-					versionLogger.Debug("Collecting images…")
-
-					// Collect images without & with Konnectivity, as Konnecctivity / OpenVPN can be switched in clusters
-					// at any time. Remove the non-Konnectivity option once OpenVPN option is finally removed.
-
-					imagesWithoutKonnectivity, err := images.GetImagesForVersion(
-						versionLogger,
-						clusterVersion,
-						cloudSpec,
-						cniPlugin,
-						false,
-						kubermaticConfig,
-						options.AddonsPath,
-						versions,
-						caBundle,
-						options.RegistryPrefix,
-					)
-					if err != nil {
-						return fmt.Errorf("failed to get images: %w", err)
-					}
-					imageSet.Insert(imagesWithoutKonnectivity...)
-
-					imagesWithKonnectivity, err := images.GetImagesForVersion(
-						versionLogger,
-						clusterVersion,
-						cloudSpec,
-						cniPlugin,
-						true,
-						kubermaticConfig,
-						options.AddonsPath,
-						versions,
-						caBundle,
-						options.RegistryPrefix,
-					)
-					if err != nil {
-						return fmt.Errorf("failed to get images: %w", err)
-					}
-					imageSet.Insert(imagesWithKonnectivity...)
+			if options.Archive && options.ArchivePath == "" {
+				currentPath, err := os.Getwd()
+				if err != nil {
+					return fmt.Errorf("failed to get current directory: %w", err)
 				}
+				options.ArchivePath = fmt.Sprintf("%s/images.tar.gz", currentPath)
```

Comment:
can we maybe change the default path here to something that is more descriptive and depends on the KKP version used? something like:

```suggestion
				options.ArchivePath = fmt.Sprintf("%s/kubermatic-v%s-images.tar.gz", currentPath, version.Kubermatic)
```